# 3) Other datasets

Use your RBF network to classify other datasets.

In [1]:
# Importing all the required stuff
import os
import random
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from rbf_net import RBF_Net

plt.style.use(['seaborn-v0_8-colorblind', 'seaborn-v0_8-darkgrid'])
plt.rcParams.update({'font.size': 20})

np.set_printoptions(suppress=True, precision=5)

# Automatically reload external modules
%load_ext autoreload
%autoreload 2



In [2]:
import numpy as np
import pandas as pd
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split

# Load the breast cancer dataset
data = load_breast_cancer()
X = data.data
y = data.target

In [3]:
# Split the dataset into training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [4]:
# Create and train the RBF network with 10 hidden units
rbf_net = RBF_Net(num_hidden_units=10, num_classes=2)
rbf_net.train(X_train, y_train)

# Predict the classes for the training and test sets
y_pred_train = rbf_net.predict(X_train)
y_pred_test = rbf_net.predict(X_test)

In [5]:
# Compute the accuracy for the training and test sets
train_acc = rbf_net.accuracy(y_train, y_pred_train)
test_acc = rbf_net.accuracy(y_test, y_pred_test)

print(f'Training accuracy: {train_acc:.2f}')
print(f'Test accuracy: {test_acc:.2f}')

Training accuracy: 0.90
Test accuracy: 0.94


# Report + Results

I have used the RBF network to classify other datasets. Specifically, I loaded the breast cancer dataset using the load_breast_cancer function from sklearn.datasets. The dataset was split into training and test sets using the train_test_split function from sklearn.model_selection. The RBF network was created with 10 hidden units and 2 output classes, and then trained on the training set using the train method from the RBF_Net class.

The predicted classes for both the training and test sets were computed using the predict method from the RBF_Net class. The accuracy of the network was then calculated using the accuracy method from the RBF_Net class. The training accuracy was found to be 0.90 and the test accuracy was 0.94.

These results show that the RBF network is able to accurately classify the breast cancer dataset. It achieves a test accuracy of 0.94, which is a strong indication of its ability to generalize well to new data.